src/darsia/measure/wasserstein.py

```diff
@@ -1587,6 +1601,17 @@ def _solve(self, flat_mass_diff: np.ndarray) -> tuple[float, np.ndarray, dict]:
                 # Assemble linear problem in Newton step
                 tic = time.time()
                 residual_i = self.residual(rhs, solution_i)
+                res_pressure = residual_i[self.pressure_slice]
+                transport_density_faces = self.transport_density_faces(flux)
+                res_opt = self.mass_matrix_faces.dot(
+                    solution_i[self.flux_slice]
+                ) - transport_density_faces * self.div.T.dot(
+                    solution_i[self.pressure_slice]
+                )
+
+                residual_opt = np.linalg.norm(res_opt, 2) / old_distance
+                residual_div = np.linalg.norm(res_pressure, 2)
+                res_pde = np.sqrt(residual_opt**2 + residual_div**2)
```

Comment:
Do I understand correctly that the extra calculation are just about how the residual norm is calculated for the convergence check but the computations of the next iterate remain the same?